modules/solid_mechanics/src/kernels/StressDivergenceRZ.C

```diff
@@ -101,11 +227,103 @@ StressDivergenceRZ::calculateJacobian( unsigned int ivar, unsigned int jvar )
   }
 
   SymmTensor tmp( _Jacobian_mult[_qp] * phi );
-  const Real val( test.doubleContraction( tmp ) );
+  Real val( test.doubleContraction( tmp ) );
 
+  // volumetric locking correction
+  // K = Bbar^T_i * C * Bbar^T_j where Bbar = B + Bvol
+  // K = B^T_i * C * B_j + Bvol^T_i * C * Bvol_j + B^T_i * C * Bvol_j + Bvol^T_i * C * B_j
+  if (_volumetric_locking_correction)
+  {
+    RealGradient new_test(2, 0.0);
+    RealGradient new_phi(2, 0.0);
+
+    new_test(0) = _grad_test[_i][_qp](0) + _test[_i][_qp] / _q_point[_qp](0);
+    new_test(1) = _grad_test[_i][_qp](1);
+    new_phi(0) = _grad_phi[_j][_qp](0) + _phi[_j][_qp] / _q_point[_qp](0);
+    new_phi(1) = _grad_phi[_j][_qp](1);
+
+    // Bvol^T_i * C * Bvol_j
+    val += _Jacobian_mult[_qp].sum_3x3() * (_avg_grad_test[_i][ivar] - new_test(ivar)) * (_avg_grad_phi[_j][jvar] - new_phi(jvar)) / 9.0;
+
+    // B^T_i * C * Bvol_j
+    RealGradient sum_3x1 = _Jacobian_mult[_qp].sum_3x1();
+    if ((ivar == 0) && (jvar == 0))
+      val += (sum_3x1(0) * test.xx() +  sum_3x1(2) * test.zz()) * (_avg_grad_phi[_j][0] - new_phi(0)) / 3.0;
+    else if ((ivar == 0) && (jvar == 1))
+      val += (sum_3x1(0) * test.xx() +  sum_3x1(2) * test.zz()) * (_avg_grad_phi[_j][1] - new_phi(1)) / 3.0;
+    else if ((ivar == 1) && (jvar == 0))
+      val += sum_3x1(1) * test.yy() * (_avg_grad_phi[_j][0] - new_phi(0)) / 3.0;
+    else
+      val += sum_3x1(1) * test.yy() * (_avg_grad_phi[_j][1] - new_phi(1)) / 3.0;
+
+    // Bvol^T_i * C * B_j
+    // tmp = C * B_j from above
+    if (ivar == 0)
+      val += (tmp.xx() + tmp.yy() + tmp.zz()) * (_avg_grad_test[_i][0] - new_test(0)) / 3.0;
+    else
+      val += (tmp.xx() + tmp.yy() + tmp.zz()) * (_avg_grad_test[_i][1] - new_test(1)) / 3.0;
+  }
   return val;
```

Comment:
A factor of 1/9 is 1/3 * 1/3, so that factor is in there. The
initialization could be made to zero with the addition of the single term
happening in the return statement. It is no big deal. The change would
remove one division operation and remove a little bit of clutter. If you
prefer to leave it like it is that's fine, too.

On Wed, Sep 7, 2016 at 8:02 PM sveerara <notifications@github.com> wrote:

> In modules/solid_mechanics/src/kernels/StressDivergenceRZ.C
> <https://github.com/idaholab/moose/pull/7531#discussion_r77935924>:
>
> > +    if ((ivar == 0) && (jvar == 0))
> > +      val += (sum_3x1(0) * test.xx() +  sum_3x1(2) * test.zz()) * (_avg_grad_phi[_j][0] - new_phi(0)) / 3.0;
> > +    else if ((ivar == 0) && (jvar == 1))
> > +      val += (sum_3x1(0) * test.xx() +  sum_3x1(2) * test.zz()) * (_avg_grad_phi[_j][1] - new_phi(1)) / 3.0;
> > +    else if ((ivar == 1) && (jvar == 0))
> > +      val += sum_3x1(1) * test.yy() * (_avg_grad_phi[_j][0] - new_phi(0)) / 3.0;
> > +    else
> > +      val += sum_3x1(1) * test.yy() * (_avg_grad_phi[_j][1] - new_phi(1)) / 3.0;
> > +
> > +    // Bvol^T_i * C * B_j
> > +    // tmp = C * B_j from above
> > +    if (ivar == 0)
> > +      val += (tmp.xx() + tmp.yy() + tmp.zz()) * (_avg_grad_test[_i][0] - new_test(0)) / 3.0;
> > +    else
> > +      val += (tmp.xx() + tmp.yy() + tmp.zz()) * (_avg_grad_test[_i][1] - new_test(1)) / 3.0;
> > +  }
> >    return val;
>
> The term val is calculated starting from line 225. The term on line 225
> does not have a factor of 1/3. The term on line 241 has a factor of 1/9.
> The other two terms are calculated in the lines 245-260 and only these two
> terms have a factor of 1/3.
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7531/files/2461c4a5d32f7ff6d53d6a3619b3f5f971929d67#r77935924>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMWPv7AonAaZ93Vctr_Yce9iOLBhL0Vks5qn2yggaJpZM4Jinz_>
> .
>